pypeit/spectrographs/p200_dbsp.py

```diff
@@ -0,0 +1,432 @@
+
+loc = EarthLocation.of_site('Palomar')
+
+def get_zenith_ra_dec(time) -> SkyCoord:
+    time = Time(time)
+    altaz = AltAz(alt=Angle(90, unit=u.deg), az=Angle(0, unit=u.deg), obstime=time, location=loc)
+    return altaz.transform_to(ICRS)
+
+class P200DBSPSpectrograph(spectrograph.Spectrograph):
+    """
+    Child to handle P200/DBSP specific code
+    """
+    ndet = 1
+
+    def __init__(self):
+        # Get it started
+        super(P200DBSPSpectrograph, self).__init__()
+        self.spectrograph = 'p200_dbsp'
+        self.telescope = telescopes.P200TelescopePar()
+
+    def configuration_keys(self):
+        """
+        Return the metadata keys that defines a unique instrument
+        configuration.
+
+        This list is used by :class:`pypeit.metadata.PypeItMetaData` to
+        identify the unique configurations among the list of frames read
+        for a given reduction.
+
+        Returns:
+            list: List of keywords of data pulled from meta
+        """
+        return ['dispname', 'binning', 'dispangle', 'dichroic']
+
+    def init_meta(self):
+        """
+        Generate the meta data dict
+        Note that the children can add to this
+
+        Returns:
+            self.meta: dict (generated in place)
+
+        """
+        meta = {}
+        # Required (core)
+        # VERY HACKY!!!!
+        meta['ra'] = dict(card=None, compound=True)
+        meta['dec'] = dict(card=None, compound=True)
+        #meta['ra'] = dict(ext=0, card='RA')
+        #meta['dec'] = dict(ext=0, card='DEC')
+        meta['target'] = dict(ext=0, card='OBJECT')
+
+        meta['dispname'] = dict(ext=0, card='GRATING')
+        meta['decker'] = dict(ext=0, card='APERTURE')
+        meta['binning'] = dict(card=None, compound=True)
+
+        meta['mjd'] = dict(card=None, compound=True)
+        meta['exptime'] = dict(ext=0, card='EXPTIME')
+        #meta['airmass'] = dict(ext=0, card='AIRMASS')
+        # VERY HACKY!!!!
+        meta['airmass'] = dict(card=None, compound=True)
+
+        # Extras for config and frametyping
+        meta['dichroic'] = dict(ext=0, card='DICHROIC')
+        meta['dispangle'] = dict(card=None, rtol=1e-2, compound=True)
+        meta['slitwid'] = dict(ext=0, card='APERTURE')
+        meta['idname'] = dict(ext=0, card='IMGTYPE')
+        # Lamps
+        meta['lampstat01'] = dict(ext=0, card='LAMPS')
+
+        # Ingest
+        self.meta = meta
+
+    def compound_meta(self, headarr: List[fits.Header], meta_key):
+        if meta_key == 'mjd':
+            return Time(headarr[0]['UTSHUT']).mjd
+        elif meta_key == 'dispangle':
+            try:
+                return Angle(headarr[0]['ANGLE']).deg
+            except Exception as e:
+                print(headarr[0]['ANGLE'])
+                raise e
+        elif meta_key == 'ra':
+            ra = headarr[0].get('RA', default=None)
+            frametype = headarr[0]['IMGTYPE']
+            if ra is None and frametype in ['bias', 'flat', 'cal']:
```

Comment:
Another option here (and maybe a better one) is to mimic what
we do for the VLT.

Reply:
Same for `dec` and `airmass`